src/providers/ipa/ipa_s2n_exop.c

```diff
@@ -1645,6 +1674,14 @@ static void ipa_s2n_get_user_done(struct tevent_req *subreq)
         state->attrs = attrs;
 
         if (attrs->response_type == RESP_USER_GROUPLIST) {
+
+            DEBUG(SSSDBG_TRACE_FUNC, "Received [%lu] groups in group list from "
+                                     "IPA Server\n", attrs->ngroups);
+
```

Comment:
ngroups is size_t so there is a -Wformat warning like:
`../src/providers/ipa/ipa_s2n_exop.c:1741:38: Fehler: Format »%lu« erwartet Argumenttyp »long unsigned int«, aber Argument 6 hat Typ »size_t {aka unsigned int}« [-Werror=format=]
        DEBUG(SSSDBG_TRACE_FUNC, "Received [%lu] groups in group list from "`

Additionally I would make c a size_t as well to avoid -Wsign-compare warnings.